crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -402,6 +403,30 @@ mod orderbook {
     // *************************************************************************
     #[generate_trait]
     impl InternalFunctions of InternalFunctionsTrait {
+        fn _fulfill_offer(ref self: ContractState, fulfill_info: FulfillInfo, order: OrderV1) {
+            if order.token_id.is_some() {
+                let (auction_order_hash, auction_end_date, auction_offer_count) = self
+                    .auctions
+                    .read(order.compute_token_hash());
+
+                assert(auction_order_hash.is_zero(), orderbook_errors::USE_FULFILL_AUCTION);
+            }
+
+            let current_date = starknet::get_block_timestamp();
+            assert(current_date >= order.start_date, orderbook_errors::OFFER_NOT_STARTED);
+            assert(order.end_date > current_date, orderbook_errors::ORDER_EXPIRED);
+
+            order_status_write(fulfill_info.order_hash, OrderStatus::Fulfilled);
+            self
+                .emit(
+                    Event::OrderFulfilled(
+                        OrderFulfilled {
+                            order_hash: fulfill_info.order_hash, fulfiller: fulfill_info.fulfiller
```

Comment:
In the case of Auction, we have a `related_order_hash` that must be present in this event. To be sync with @kwiss in the PR #187 if this must be done here first or in auction first.

In the case of the classic offers, the `related_order_hash` must be `Option::None`.